### PR TITLE
fixed segfault when a BlockUniTensor with no valid blocks was printed…

### PR DESCRIPTION
**Pull request by manuschneider**
Fixed a segfault in case no valid block exists. For example, the following code resulted in a kernel crash:
```
bond_d = cytnx.Bond(
     cytnx.BD_IN, [cytnx.Qs(5)>>1, cytnx.Qs(-5)>>1],
     [cytnx.Symmetry.U1()]
)
bond_e = cytnx.Bond(
     cytnx.BD_IN, [cytnx.Qs(1)>>1, cytnx.Qs(-1)>>1],
     [cytnx.Symmetry.U1()]
)
bond_f = cytnx.Bond(
     cytnx.BD_OUT,
     [cytnx.Qs(2)>>1, cytnx.Qs(0)>>2, cytnx.Qs(-2)>>1],
     [cytnx.Symmetry.U1()]
)

Tsymm = cytnx.UniTensor([bond_d, bond_e, bond_f], name="symm. tensor",
labels=["d", "e", "f"])

Tsymm.print_diagram() 
```

### CODE REVIEW

File: include/UniTensor.hpp [hunk 1]
```diff
@@ -1171,28 +1171,30 @@ namespace cytnx {
...
-      return this->_blocks[0].dtype();
+      return this->_blocks.size() < 1 ? N_Type + 1 : this->_blocks[0].dtype();
     };
     int device() const {
   #ifdef UNI_DEBUG
       cytnx_error_msg(this->_blocks.size() == 0, "[ERROR][internal] empty blocks for blockform.%s",
                       "\n");
   #endif
-      return this->_blocks[0].device();
+      return this->_blocks.size() < 1 ? 404 : this->_blocks[0].device();
```
Comment (IvanaGyro) [inline include/UniTensor.hpp:1181]: Do we assume the device value larger and equal to 0 is a GPU device?

File: include/UniTensor.hpp [hunk 2]
```diff
@@ -1171,28 +1171,30 @@ namespace cytnx {
       cytnx_error_msg(this->_blocks.size() == 0, "[ERROR][internal] empty blocks for blockform.%s",
                       "\n");
   #endif
-      return this->_blocks[0].dtype();
+      return this->_blocks.size() < 1 ? N_Type + 1 : this->_blocks[0].dtype();
```
Comment (IvanaGyro) [inline include/UniTensor.hpp:1174]: Is `Type.Void` a choice here?

**Comment by ianmccul** [comment]
It worries me that it is doing something completely different in UNI_DEBUG mode (displaying an error, and throwing an exception) versus non-debug mode (return a normal value).

File: include/UniTensor.hpp (hunk 1 above)
Comment (kaihsin) [reply include/UniTensor.hpp:1181]: Yes. See Device.hpp

**Comment by kaihsin** [comment]
> It worries me that it is doing something completely different in UNI_DEBUG mode (displaying an error, and throwing an exception) versus non-debug mode (return a normal value).

It is legacy from uni10 where we have two builds for test and performances. I don't think it's necessary anymore.

**Comment by manuschneider** [comment]
I updated the standard values, thanks @IvanaGyro for checking.

**Review by IvanaGyro — APPROVED** [review]

**Pull request merged**